src/com/facebook/buck/jvm/groovy/GroovyConfiguredCompilerFactory.java

```diff
@@ -16,24 +16,36 @@
 
 package com.facebook.buck.jvm.groovy;
 
+import com.facebook.buck.core.model.BuildTarget;
 import com.facebook.buck.core.rules.BuildRuleResolver;
 import com.facebook.buck.core.toolchain.ToolchainProvider;
+import com.facebook.buck.core.toolchain.tool.Tool;
 import com.facebook.buck.jvm.groovy.GroovyLibraryDescription.CoreArg;
 import com.facebook.buck.jvm.java.CompileToJarStepFactory;
 import com.facebook.buck.jvm.java.ConfiguredCompilerFactory;
 import com.facebook.buck.jvm.java.JavacOptions;
 import com.facebook.buck.jvm.java.JvmLibraryArg;
+import com.facebook.buck.util.Optionals;
+import com.google.common.collect.ImmutableCollection;
 import java.util.Objects;
 import java.util.Optional;
 import javax.annotation.Nullable;
 
 public class GroovyConfiguredCompilerFactory extends ConfiguredCompilerFactory {
-  private final GroovyBuckConfig groovyBuckConfig;
+  @Nullable private final GroovyBuckConfig groovyBuckConfig;
+  @Nullable private Tool groovyc;
 
-  public GroovyConfiguredCompilerFactory(GroovyBuckConfig groovyBuckConfig) {
+  public GroovyConfiguredCompilerFactory(@Nullable GroovyBuckConfig groovyBuckConfig) {
     this.groovyBuckConfig = groovyBuckConfig;
   }
 
+  private Tool getGroovyc() {
```

Comment:
This needs to be annotated with `@Nullable` since it can return `null` when both `groovyc` and `groovyBuckConfig` is `null`.